src/SAWScript/MGU.hs

```diff
@@ -382,10 +387,13 @@ instance Instantiate Type where
     TySkolemVar _ _ -> ty
     LType pos ty'   -> LType pos (instantiate nts ty')
 
+instantiateType :: Instantiate t => Map Name Type -> t -> t
+instantiateType tenv = instantiate (M.assocs tenv)
```

Comment:
This does almost the exact same thing as the `instantiate` method of the `Instantiate` class, except that this function uses a `Map Name Type` instead of `[(Name, Type)]`. I think using a `Map` makes more sense here—any objection to simply changing the type signature of `instantiate` to use a `Map` instead?

Either way, it would be worth leaving a comment saying precisely what we are instantiating. (Particularly, we're instantiating typedefs.)

Reply:
Ok, I've changed `instantiate` to use `Map`. It seems like it's not only used with typedefs, so I wrote a more general description.